src/domain/transaction/evm/price_calculator.rs

```diff
@@ -138,11 +152,19 @@ impl<G: EvmGasPriceServiceTrait> PriceCalculator<G> {
                 relayer,
             )?;
 
+        let extra_fee: u128 = if let Some(service) = &self.network_extra_fee_calculator_service {
```

Comment:
We can use Option here as well given that in line 167 we are wrapping it in some.

so for example default value can simply be None instead of 0